### PR TITLE
feat: add Lua_nvim interpreter

### LINKED ISSUE #5: Lua_nvim support
Hey ! Thanks for this awesome plugin !

I think it can be really useful to execute lua snippets within neovim, how hard do you think it would be ?
And how could I help in the process ?

Thanks again !

### PR DESCRIPTION
**Pull request by vigoux**
This interpreter will run lua chuncks using nvim lua executor.

It feels weird to not be able to execute this directly wihtin neovim though, as I am opening a new nvim instance each time.

Fixes #5

### CODE REVIEW

**Comment by michaelb** [comment]
Hi, you don't have to "open a new nvim instance each time"

If that's pure lua code you are executing, just call the lua interpreter that's on your system with the file you've just written to as argument, no need to call it through nvim

Also, you don't need nor want to modify src/interpreters/mod.rs the file, just create  your Lua_nvim.rs file and it magically works :-)

File: src/interpreters/Lua_nvim.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    fn build(&mut self) -> Result<(), SniprunError> {
+        let mut _file =
+            File::create(&self.main_file_path).expect("Failed to create file for lua-nvim");
+
+        write(&self.main_file_path, &self.code).expect("Unable to write to file for lua-nvim");
+        Ok(())
+    }
+
+    fn execute(&mut self) -> Result<String, SniprunError> {
+        let output = Command::new("nvim")
```
Comment (michaelb) [inline src/interpreters/Lua_nvim.rs:86]: why not just Command::new("lua").arg(&self.main_file_path)..... ? that should do the job

**Comment by vigoux** [comment]
> If that's pure lua code you are executing, just call the lua interpreter that's on your system with the file you've just written to as argument, no need to call it through nvim

I actually want that to execute nvim-related code :wink: 

> Also, you don't need nor want to modify src/interpreters/mod.rs the file is generated automatically at each build, just create your Lua_nvim.rs file and it magically works :-)

I did not edited anything in the `mod.rs`, that's just what the script did :wink:

File: src/interpreters/Lua_nvim.rs (hunk 1 above)
Comment (vigoux) [reply src/interpreters/Lua_nvim.rs:86]: Because I want that to execute within neovim :wink:

Comment (vigoux) [reply src/interpreters/Lua_nvim.rs:86]: I even want that toexecute within the very instance that is running the plugin, but I don't know how to do that.

Comment (michaelb) [reply src/interpreters/Lua_nvim.rs:86]: I can't easily give you the nvim handle (due to ownership reasons in rust) from sniprun , but I can maybe execute commands you'd pass me back as a result (with a special identifier so I know whether to echo ":split" or to :split)

Comment (vigoux) [reply src/interpreters/Lua_nvim.rs:86]: > with a special identifier so I know whether to echo ":split" or to :split

I dont get that

Comment (michaelb) [reply src/interpreters/Lua_nvim.rs:86]: usually you would return Ok(the stdout of the  code), but I can maybe arrange to have an Ok(magic number + command to run) or maybe an Err variant bring back a command to execute, such as ':split' or ':w' for example

**Comment by michaelb** [comment]
does 'nvim-related' lua code have a different filetype than just pure lua? 

Sniprun relies on detected filetype to run the correct interpreter, and I can't yet manage multiple interpreter by filetype

As I noticed that the lua_nvim interpreter may have trouble processing pure lua code when you are calling it as your Command specify

File: src/interpreters/Lua_nvim.rs (hunk 1 above)
Comment (vigoux) [reply src/interpreters/Lua_nvim.rs:86]: If I can have the nvim handle, that can should be perfect, how about passing it as an argument to `execute` ?

Comment (michaelb) [reply src/interpreters/Lua_nvim.rs:86]: i'll have a look but I can't promise it's possible

**Comment by michaelb** [comment]
> Got it working in the end, but this is not executing in current nvim instance, but in a fresh instance each time.

Nice, I don't think getting you the current neovim instance is probably impossible (for me) because passing the nvim instance is a ownership & lifetime nightmare, and has repercussion through many, many things

Could you copy paste an example of lua-nvim code? (the chunk you would 'sniprun'), it'll help me integrate your interpreter alongside the 'real' lua one

**Comment by michaelb** [comment]
(a test example please, since i don't know how to test this)

**Comment by vigoux** [comment]
Just run `print(vim.api.nvim_get_current_buf())` for example, it should work

**Comment by michaelb** [comment]
> Just run `print(vim.api.nvim_get_current_buf())` for example, it should work

got it, works greats! I'll see by myself to manage 'lua-nvim' and 'lua' differenciation

**Review by michaelb — APPROVED** [review]

**Pull request merged**